common/lib/xmodule/xmodule/modulestore/split_mongo/split_draft.py

```diff
@@ -198,6 +198,15 @@ def delete_item(self, location, user_id, revision=None, **kwargs):
                 branches_to_delete = [ModuleStoreEnum.BranchName.published, ModuleStoreEnum.BranchName.draft]
             elif revision is None:
                 branches_to_delete = [ModuleStoreEnum.BranchName.draft]
+                draft_location = location.for_branch(ModuleStoreEnum.BranchName.draft)
+                try:
+                    item = self.get_item(draft_location)
+                    if getattr(item, 'has_children', False):
+                        # If item have has_published_version then delete published children also.
+                        if self.has_published_version(item):
+                            branches_to_delete.insert(0, ModuleStoreEnum.BranchName.published)
+                except ItemNotFoundError:
```

Comment:
@adampalay , @awaisdar001 BlockKey(type='about', id='syllabus') is not in the fixture of the course and so it throwing ItemNotFoundError and so new question is why this is failing in this current pr ? The answer is we are using item = self.get_item(draft_location) in the current branch and its not present in master. So other solution is this can be resolve in the fixtures of course by adding this block_key but in one case what happened if item is not present and we hit the function delete_item ? it will break the functionality using current logic so catching the ItemNotFoundError exception will protect this logic without breaking.